pkg/util/wait/backoff.go

```diff
@@ -0,0 +1,73 @@
+package wait
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/utils/clock"
+	"k8s.io/utils/ptr"
+)
+
+// UntilWithBackoff runs f in a loop until context indicates finished. It
+// applies backoff depending on the SpeedSignal f returns.  Backoff increases
+// exponentially, ranging from 1ms to 100ms.
+func UntilWithBackoff(ctx context.Context, f func(context.Context) SpeedSignal) {
+	mgr := defaultBackoffManager()
+	wait.BackoffUntil(func() {
+		mgr.toggleBackoff(f(ctx))
+	}, mgr, false, ctx.Done())
+}
+
+// SpeedSignal indicates whether we should run the function again immediately,
+// or apply backoff.
+type SpeedSignal interface{}
+
+// KeepGoing signals to continue immediately.
+type KeepGoing struct{ SpeedSignal }
+
+// SlowDown signals to backoff.
+type SlowDown struct{ SpeedSignal }
+
+func (s *speedyBackoffManager) toggleBackoff(speedSignal SpeedSignal) {
+	switch speedSignal.(type) {
+	case KeepGoing:
+		if *s.backingOff {
+			*s.backingOff = false
+			s.rateLimiter.Forget("")
+		}
+	case SlowDown:
+		*s.backingOff = true
+	}
+}
+
+type speedyBackoffManager struct {
+	backingOff  *bool
+	rateLimiter workqueue.RateLimiter
+	timer       clock.Timer
+	wait.BackoffManager
```

Comment:
```suggestion
```

I'm not really a fan of this style since reviewers often miss if this struct (`speedBackoffManager`) implements any interfaces since we're not sure if this is a struct or interface like this: https://github.com/kubernetes-sigs/kueue/pull/2102/files#r1594467180

So, I would recommend using `var _ wait.BackoffManager =(*speedyBackoffManager)(nil)` as mentioned above.